### DISCUSSION TITLE
Laravel backpack pro relationship field: add the same child twice in the same parent cant be limited

### DISCUSSION THREAD

**Discussion by gillesboisson** [Laravel-Backpack/community-forum#498] [Q&A (Help)] [answered]
is there any way on relationship Pro fields to not allow having the same instance twice as a children in a belongsToMany relationship. Is there any to have some kind of unique constrain

example here on stackoverflow:

![Screenshot](https://i.stack.imgur.com/7l3gC.png)

I can add the same POI twice.

I tried to look on the doc and check the field blade view, I didn't find any option to filter the dropdown list

**Comment by pxpm** [comment]
Hey @gillesboisson 

You can configure the "pivot select" using the `pivotSelect` attribute in field definition: https://backpackforlaravel.com/docs/5.x/crud-fields#save-additional-data-to-pivot-table (scroll a little bit down). 

You can make it an ajax field and then filter the previously selected options by sending the other form values with: `include_all_form_fields => true` on the `pivotSelect` configuration. 

You can then intercept the request on the in ajax endpoint (You can use `FetchOperation` for that https://backpackforlaravel.com/docs/5.x/crud-operation-fetch#how-to-use-1) and filter out previously selected values.

There are plans to make this "easier" by having some option developers can set on the field definition like `singlePivotSelection => true/false` but it's not currently implemented, and not on the road map to happen before v6 launch. 

Let me know how it goes.

Cheers

**Comment by gillesboisson** [comment, marked as answer]
Thanks for your quick feedback, It works perfectly. Here is an example of my implementation 

```php
// Fetch implementation
  public function fetchPois(PoiListRequest $request)
  {

    // get all linked uuids from the request
    $forms = $request->input('form');
    $uuids = [];

    foreach ($forms as $form) {
      $formName = $form['name'];
      if ($formName === 'uuid') {
        $uuids[] = $form['value'];
      }
    }

    return $this->fetch([
      'model' => Poi::class, // required
      'query' => function ($model) use ($uuids) {
        return $model
          ->whereNotIn('uuid', $uuids);
      }
    ]);
  }

// ...

protected function setupUpdateOperation()
{
   // ...
       $this->crud->addField([
      'tab' => trans($langPrefix . 'tabs.pois'),
      'name'          => 'pois',
      'type'          => "relationship",   
      'allows_multiple' => false,
      'subfields'   => [
        [
          'name' => 'trigger_radius',
          'type' => 'number',
        ],
      ],
      // allow to ignore poi which has already been added to the hub map
      'pivotSelect' => [
        "include_all_form_fields" => true,
        'ajax' => true,
        'placeholder' => 'Pick a poi',
        'wrapper' => [
          'class' => 'col-md-6',
        ],

      ],
    ]);
    // ...  

```

```php
// route setup
  Route::prefix('/hubmap')->group(function () {
    Route::post('/fetch/pois', 'HubMapCrudController@fetchPois');   // pivot select poi fetch
    Route::crud('/', 'HubMapCrudController');
  });
});
```

**Reply by pxpm** [reply]
Thank you so much for providing the example code. I am sure it will help other devs too. Much appreciated 🙏 

Let me know if I can help you with something else. 

Cheers
